codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -74,6 +77,9 @@ public HttpHeaders add(HttpHeaders headers) {
     @Override
     public HttpHeaders set(HttpHeaders headers) {
         if (headers instanceof DefaultHttpHeaders) {
+            if (headers == this) {
+                throw new IllegalArgumentException("can't add to itself.");
```

Comment:
I'm leaning toward throwing an exception in both cases, since it is a programming error and you want to be explicit.  Not throwing may lead to buggy application code.   That said, I don't feel strongly.